enhancements/storage/csi-driver-global-configmap-secret-mounter.md

```diff
@@ -0,0 +1,201 @@
+---
+title: global-configmap-secret-mounter
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Global ConfigMap and Secret Mounter
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+ 1.  It is possible to build some sorts of per-namespace ACL boundaries if we really wanted to, but I think it makes more sense
+ to have the idea of global or namespaced and no in-between since nothing else is in between today.  They can write a controller
+ if they really want to create resources shared across some namespaces and not others.
+
+## Summary
+
+A CSI driver that can mount a configmap or secret from a specified namespace into any pod.
+This would allow a single copy of a configmap to exist in openshift-config-managed (say the trusted-ca) and mount that configmap into any pod.
+There is no ACL check on the pod attempting to mount it.  Any configmap or secret that can be mounted, can be mounted anywhere.
+The CSI driver must keep the content on disk up to date, similar to how the kubelet keeps configmap and secret data consistent today.
+This allows the creation of a webhook admission plugin to allow annotation based injection of well-known configmaps and secrets, like the trusted-ca today.
+
+To have reasonable security boundaries, the cluster-admin must explicitly enumerate all namespace/name tuples for the configmaps and secrets that he wants to expose. 
+This will allow the CSI driver to use individual list/watches to drive the watch behavior to have scoped RBAC for the CSI driver.
```

Comment:
This sentence is a bit vague. The CSI driver will run a pod on every node (via daemonset). Will all of them watch all namespace/name tuples? The number of watches can grow quickly.

If the number of pods that use such volumes is smaller than number of nodes, the CSI driver could keep a watch for every mounted secret / config map.

Reply:
> If the number of pods that use such volumes is smaller than number of nodes, the CSI driver could keep a watch for every mounted secret / config map.

An implementation could use "min of" sorts of logic.  Kubelets already place a large watch load on configmaps and secrets, so a few more per node doesn't cause me much heartburn.